account_invoice_split/wizard/account_invoice_split.py

```diff
@@ -0,0 +1,194 @@
+import openerp.addons.decimal_precision as dp
+
+FIELD_MAPPING = [('id', 'origin_invoice_line_id'),
+                 ('product_id', 'product_id'),
+                 ('name', 'name'),
+                 ('quantity', 'total_quantity')]
+
+
+class AccountInvoiceSplit(models.TransientModel):
+    _name = 'account.invoice.split'
+
+    line_ids = fields.One2many(
+        comodel_name='account.invoice.split.line', inverse_name='wizard_id',
+        string='Invoice lines to Split')
+
+    @api.model
+    def _check_condition(self):
+        active_ids = self.env.context.get('active_ids')
+        assert len(active_ids) == 1
+        invoice = self.env['account.invoice'].browse(active_ids)[0]
+        if invoice.state != 'draft':
+            raise exceptions.Warning(_('Invoice must be draft not %s') %
+                                     (invoice.state,))
+
+    @api.model
+    def default_get(self, fields_list):
+        self._check_condition()
+        res = super(AccountInvoiceSplit, self).default_get(self)
+        active_ids = self.env.context.get('active_ids')
+        assert len(active_ids) == 1
+        invoice = self.env['account.invoice'].browse(active_ids)[0]
+        split_lines = []
+        for line in invoice.invoice_line:
+            current_line_values = {
+                'origin_invoice_line_id': line.id,
+                'product_id': line.product_id.id,
+                'name': line.name,
+                'total_quantity': line.quantity,
+            }
+            split_lines.append((0, 0, current_line_values))
+        res['line_ids'] = split_lines
+        return res
+
+    @api.model
+    def _get_invoice_values(self, invoice):
+        """Return all necessary values to create the new invoice.
+        Can be override to add some fields"""
+        return {
+            'origin': '%s' % (invoice.origin or '',),
+            'partner_id': invoice.partner_id.id,
+            'journal_id': invoice.journal_id.id,
+            'user_id': invoice.user_id.id,
+            'currency_id': invoice.currency_id.id,
+            'company_id': invoice.company_id.id,
+            'type': invoice.type,
+            'account_id': invoice.account_id.id,
+            'state': 'draft',
+            'reference': '%s' % (invoice.reference or '',),
+            'name': '%s' % (invoice.name or '',),
+            'fiscal_position': invoice.fiscal_position.id,
+            'payment_term': invoice.payment_term.id,
+            'period_id': invoice.period_id.id,
+            'invoice_line': [],
+            'partner_bank_id': invoice.partner_bank_id.id,
+            'date_invoice': invoice.date_invoice,
+            'date_due': invoice.date_due,
+        }
+
+    @api.model
+    def _create_invoice(self, vals):
+        new_invoice = False
+        if vals['invoice_line']:
+            new_invoice = self.env['account.invoice'].create(vals)
+        if not new_invoice:
+            raise exceptions.Warning(_('Nothing to split'))
```

Comment:
Can you write: "There is nothing to split. Please fill the 'quantities to split' column." This would be more explicit for the user.